src/actions/index.js

```diff
@@ -0,0 +1,26 @@
+import sampleData from './sampleData';
+
+/**
+ * Makes the mock "API request"
+ */
+const fetchProducts = () => {
+  return new Promise((resolve) => {
+    window.setTimeout(() => {
+      resolve(sampleData.products);
+    }, 1000);
+  });
+}
+
+/**
+ * Retrieves product list from mock API
+ */
+export const fetchProductsFromAPI = () => {
+  return (dispatch) => {
+    return fetchProducts().then(
+      (resp) => dispatch({
+        type: 'UPDATE_PRODUCTS',
+        products: resp
```

Comment:
Check out [flux standard actions](https://github.com/redux-utilities/flux-standard-action) for a common standard for structuring actions.